src/cirq_iqm/iqm_sampler.py

```diff
@@ -140,14 +146,17 @@ def run_iqm_batch(self, programs: list[cirq.Circuit], repetitions: int = 1) -> l
         for program in programs:
             self._device.validate_circuit(program)
 
-        measurements = self._send_circuits(programs, repetitions=repetitions)
-        return [study.ResultDict(measurements=meas) for meas in measurements]
+        circuit_results = self._send_circuits(
+            programs,
+            repetitions=repetitions,
+        )
+        return [IQMResult(measurements=result, metadata=metadata) for result, metadata in circuit_results]
 
     def _send_circuits(
         self,
         circuits: list[cirq.Circuit],
         repetitions: int = 1,
-    ) -> list[dict[str, np.ndarray]]:
+    ) -> list[tuple[dict[str, np.ndarray], ResultMetadata]]:
         """Sends a batch of circuits to be executed and retrieves the results.
 
         If a user interrupts the program while it is waiting for results, attempts to abort the submitted job.
```

Comment:
The docstring could have Args: and Returns: fields here

Reply:
Changed